cloudwatch_logger/src/log_node_param_helper.cpp

```diff
@@ -198,6 +192,19 @@ Aws::AwsError ReadSubscriberList(
   return ret;
 }
 
+Aws::AwsError ReadIgnoreNodesSet(
```

Comment:
Other parameter reading methods check `ret` and use a default value if the parameter is missing or there is a read error, and emit a log accordingly. This read method should do the same for consistency

Reply:
This also implies adding a corresponding test to log_node_param_helper_test.cpp

Besides that, the new code looks good to me!

Reply:
Added an error log message if an unexpected exception occurred. I did not do anything if it was not found because it is an optional parameter that's okay to not include and the default in this case is an empty list. 

Also added a unit test.

Reply:
Looks good to me. The only thing is that we are not adding a default for `ignore_nodes` when that config value is not set. That means customers that upgrade the node, and that use their own configuration file, will see a different behaviour because that config file won't be updated. I think we can get a better upgrade experience by setting `["/cloudwatch_logger", "/cloudwatch_metrics_collector"]` as the default on `ReadIgnoreNodesSet`, both for param read with `AWS_ERR_NOT_FOUND` and as a basic recovery for param read with another error. What do you think?

Reply:
I wanted the default behavior to be very obvious. To me it is more intuitive that the default is an empty list and that if I don't include it then my node doesn't filter anything. 

I agree it is a slightly more jarring upgrade experience, but my opinion is that it is a better new user experience.